tests/cpp_unit_tests/test_pgm_vnf_converter.cpp

```diff
@@ -2,12 +2,92 @@
 //
 // SPDX-License-Identifier: MPL-2.0
 
+#include <power_grid_model_io_native/common/enum.hpp>
 #include <power_grid_model_io_native/vnf_converter/vnf_pgm_converter.hpp>
 
+#include <power_grid_model/auxiliary/dataset.hpp>
+#include <power_grid_model/auxiliary/meta_data.hpp>
+#include <power_grid_model/auxiliary/meta_gen/gen_getters.hpp>
+
+#include <cstring>
 #include <doctest/doctest.h>
 
 namespace power_grid_model_io_native {
 
-TEST_CASE("Test converter constructor") { CHECK_THROWS_AS(PgmVnfConverter(), power_grid_model::ExperimentalFeature); };
+using enum ExperimentalFeatures;
+
+std::string_view json_string = R"({"version":"1.0","type":"input","is_batch":false,"attributes":{},"data":{}})";
+
+TEST_CASE("Test converter constructor") {
+    SUBCASE("Without experimental features enabled") {
+        CHECK_THROWS_AS(PgmVnfConverter("", experimental_features_disabled), power_grid_model::ExperimentalFeature);
+    }
+
+    SUBCASE("With experimental features enabled") { CHECK_NOTHROW(PgmVnfConverter("", experimental_features_enabled)); }
+}
+
+TEST_CASE("Test if the parse_vnf_file function can be called") {
+    auto converter = PgmVnfConverter("", experimental_features_enabled);
+    CHECK_NOTHROW(converter.parse_vnf_file());
+}
+
+TEST_CASE("Test the convert_input function") {
```

Comment:
this entire file: shorter is better
```suggestion
TEST_CASE("Test convert_input") {
```